engine/storage/datamotion/src/org/apache/cloudstack/storage/motion/StorageSystemDataMotionStrategy.java

```diff
@@ -172,78 +232,211 @@ private void validate(SnapshotInfo snapshotInfo) {
+        CopyCmdAnswer copyCmdAnswer = null;
+        String errMsg = null;
+
+        TemplateInfo templateInfo = (TemplateInfo)srcData;
+        VolumeInfo volumeInfo = (VolumeInfo)destData;
+
+        HostVO hostVO = getHost(volumeInfo.getDataCenterId(), true);
+
+        if (hostVO == null) {
+            throw new CloudRuntimeException("Unable to locate a host capable of resigning in the zone with the following ID: " + volumeInfo.getDataCenterId());
+        }
+
+        boolean computeClusterSupportsResign = computeClusterSupportsResign(hostVO.getClusterId());
+
+        if (!computeClusterSupportsResign) {
+            throw new CloudRuntimeException("Unable to locate an applicable host with which to perform a resignature operation");
+        }
+
+        try {
+            VolumeDetailVO volumeDetail = new VolumeDetailVO(volumeInfo.getId(),
+                    "cloneOfTemplate",
+                    String.valueOf(templateInfo.getId()),
+                    false);
+
+            volumeDetail = _volumeDetailsDao.persist(volumeDetail);
+
+            AsyncCallFuture<VolumeApiResult> future = _volumeService.createVolumeAsync(volumeInfo, volumeInfo.getDataStore());
+            VolumeApiResult result = future.get();
+
+            if (volumeDetail != null) {
+                _volumeDetailsDao.remove(volumeDetail.getId());
+            }
+
+            if (result.isFailed()) {
+                s_logger.debug("Failed to create a volume: " + result.getResult());
+
+                throw new CloudRuntimeException(result.getResult());
+            }
+
+            volumeInfo = _volumeDataFactory.getVolume(volumeInfo.getId(), volumeInfo.getDataStore());
+
+            volumeInfo.processEvent(Event.MigrationRequested);
+
+            volumeInfo = _volumeDataFactory.getVolume(volumeInfo.getId(), volumeInfo.getDataStore());
+
+            copyCmdAnswer = performResignature(volumeInfo, hostVO);
+
+            if (copyCmdAnswer == null || !copyCmdAnswer.getResult()) {
+                if (copyCmdAnswer != null && !StringUtils.isEmpty(copyCmdAnswer.getDetails())) {
+                    throw new CloudRuntimeException(copyCmdAnswer.getDetails());
                 }
                 else {
-                    snapshotInfo.processEvent(Event.OperationFailed);
+                    throw new CloudRuntimeException("Unable to perform host-side operation");
                 }
             }
-            catch (Exception ex) {
-                s_logger.debug(ex.getMessage(), ex);
+        } catch (Exception ex) {
+            try {
+                volumeInfo.getDataStore().getDriver().deleteAsync(volumeInfo.getDataStore(), volumeInfo, null);
+            } catch (Exception e) {
+                s_logger.debug("Exception while deleting volume on the backend " + e.getMessage());
             }
+
+            throw new CloudRuntimeException("Create volume from template failed " + ex.getMessage());
```

Comment:
Please add the template ID to the error message to assist operational debugging.